pkg/providers/instancetype/instancetype.go

```diff
@@ -93,13 +97,106 @@ func (p *DefaultProvider) List(ctx context.Context, kc *v1alpha1.KubeletConfigur
 }
 
 func (p *DefaultProvider) UpdateInstanceTypes(ctx context.Context) error {
+	p.muInstanceTypeInfo.Lock()
+	defer p.muInstanceTypeInfo.Unlock()
+
+	instanceTypes, err := getAllInstanceTypes(p.ecsClient)
+	if err != nil {
+		return err
+	}
+
+	if p.cm.HasChanged("instance-types", instanceTypes) {
+		// Only update instanceTypesSeqNun with the instance types have been changed
+		// This is to not create new keys with duplicate instance types option
+		atomic.AddUint64(&p.instanceTypesSeqNum, 1)
+		log.FromContext(ctx).WithValues(
+			"count", len(instanceTypes)).V(1).Info("discovered instance types")
+	}
+	p.instanceTypesInfo = instanceTypes
```

Comment:
Maybe put the mutex lock here? It will cost much time to finish `getAllInstanceTypes`

Reply:
I don't think it's necessary, this function is executed rarely. 
on the other hand：
> // DO NOT REMOVE THIS LOCK ----------------------------------------------------------------------------
	// We lock here so that multiple callers to getInstanceTypeOfferings do not result in cache misses and multiple
	// calls to EC2 when we could have just made one call.